cpp/oneapi/dal/backend/primitives/rng/test/rng_dpc.cpp

```diff
@@ -0,0 +1,216 @@
+
+#include "oneapi/dal/test/engine/common.hpp"
+#include "oneapi/dal/test/engine/fixtures.hpp"
+#include "oneapi/dal/test/engine/dataframe.hpp"
+
+#include "oneapi/dal/backend/primitives/rng/device_engine.hpp"
+#include "oneapi/dal/backend/primitives/rng/host_engine.hpp"
+#include "oneapi/dal/backend/primitives/rng/rng_types.hpp"
+#include "oneapi/dal/backend/primitives/ndarray.hpp"
+#include <vector>
+
+#include "oneapi/dal/backend/primitives/rng/utils.hpp"
+#include "oneapi/dal/table/common.hpp"
+
+namespace oneapi::dal::backend::primitives::test {
+
+namespace te = dal::test::engine;
+
+class mt2203 {};
+class mcg59 {};
+class mrg32k3a {};
+class mt19937 {};
+class philox4x32x10 {};
+
+template <typename engine_type>
+struct engine_map {};
+
+template <>
+struct engine_map<mt2203> {
+    constexpr static auto value = engine_type::mt2203;
+};
+
+template <>
+struct engine_map<mcg59> {
+    constexpr static auto value = engine_type::mcg59;
+};
+
+template <>
+struct engine_map<mrg32k3a> {
+    constexpr static auto value = engine_type::mrg32k3a;
+};
+
+template <>
+struct engine_map<philox4x32x10> {
+    constexpr static auto value = engine_type::philox4x32x10;
+};
+
+template <>
+struct engine_map<mt19937> {
+    constexpr static auto value = engine_type::mt19937;
+};
+
+template <typename engine_type>
+constexpr auto engine_v = engine_map<engine_type>::value;
+
+template <typename TestType>
+class rng_test : public te::float_algo_fixture<std::tuple_element_t<0, TestType>> {
+public:
+    using DataType = std::tuple_element_t<0, TestType>;
+    using EngineType = std::tuple_element_t<1, TestType>;
+    static constexpr auto engine_test_type = engine_v<EngineType>;
+
+    auto get_host_engine(std::int64_t seed) {
+        auto rng_engine = host_engine(seed, engine_test_type);
+        return rng_engine;
+    }
+
+    auto get_device_engine(std::int64_t seed) {
+        auto rng_engine = device_engine(this->get_queue(), seed, engine_test_type);
+        return rng_engine;
+    }
+
+    auto allocate_array_host(std::int64_t elem_count) {
+        auto arr_host = ndarray<DataType, 1>::empty({ elem_count });
+        return arr_host;
+    }
+
+    auto allocate_array_device(std::int64_t elem_count) {
+        auto& q = this->get_queue();
+        auto arr_gpu = ndarray<DataType, 1>::empty(q, { elem_count }, sycl::usm::alloc::device);
+        return arr_gpu;
+    }
+
+    void check_results(const ndarray<DataType, 1>& arr_1, const ndarray<DataType, 1>& arr_2) {
+        const auto arr_1_host = arr_1.to_host(this->get_queue());
+        const DataType* val_arr_1_host_ptr = arr_1_host.get_data();
+
+        const auto arr_2_host = arr_2.to_host(this->get_queue());
+        const DataType* val_arr_2_host_ptr = arr_2_host.get_data();
+
+        for (std::int64_t el = 0; el < arr_2_host.get_count(); el++) {
+            // Due to MKL inside generates floats on GPU and doubles on CPU, it makes sense to add minor eps.
```

Comment:
maybe eps should be a function arg? also what is the range of values for results here - 0.01 seems a bit large

Reply:
More likely its due to mismatch inside mkl, and its not possible to somehow fix it on oneDAL side